deepface/modules/demography.py

```diff
@@ -130,83 +130,107 @@ def analyze(
+        # Check if the face content is empty
         if img_content.shape[0] == 0 or img_content.shape[1] == 0:
             continue
 
-        # rgb to bgr
+        # Convert the image to RGB format from BGR
         img_content = img_content[:, :, ::-1]
-
-        # resize input image
+        # Resize the image to the target size for the model
         img_content = preprocessing.resize_image(img=img_content, target_size=(224, 224))
 
-        obj = {}
-        # facial attribute analysis
-        pbar = tqdm(
-            range(0, len(actions)),
-            desc="Finding actions",
-            disable=silent if len(actions) > 1 else True,
-        )
-        for index in pbar:
-            action = actions[index]
-            pbar.set_description(f"Action: {action}")
-
-            if action == "emotion":
-                emotion_predictions = modeling.build_model(
-                    task="facial_attribute", model_name="Emotion"
-                ).predict(img_content)
-                sum_of_predictions = emotion_predictions.sum()
-
-                obj["emotion"] = {}
-                for i, emotion_label in enumerate(Emotion.labels):
-                    emotion_prediction = 100 * emotion_predictions[i] / sum_of_predictions
-                    obj["emotion"][emotion_label] = emotion_prediction
-
-                obj["dominant_emotion"] = Emotion.labels[np.argmax(emotion_predictions)]
-
-            elif action == "age":
-                apparent_age = modeling.build_model(
-                    task="facial_attribute", model_name="Age"
-                ).predict(img_content)
-                # int cast is for exception - object of type 'float32' is not JSON serializable
-                obj["age"] = int(apparent_age)
-
-            elif action == "gender":
-                gender_predictions = modeling.build_model(
-                    task="facial_attribute", model_name="Gender"
-                ).predict(img_content)
-                obj["gender"] = {}
-                for i, gender_label in enumerate(Gender.labels):
-                    gender_prediction = 100 * gender_predictions[i]
-                    obj["gender"][gender_label] = gender_prediction
+        valid_faces.append(img_content)
+        face_regions.append(img_obj["facial_area"])
+        face_confidences.append(img_obj["confidence"])
 
-                obj["dominant_gender"] = Gender.labels[np.argmax(gender_predictions)]
+    # If no valid faces are found, return an empty list
+    if not valid_faces:
+        return []
 
-            elif action == "race":
-                race_predictions = modeling.build_model(
-                    task="facial_attribute", model_name="Race"
-                ).predict(img_content)
-                sum_of_predictions = race_predictions.sum()
+    # Convert the list of valid faces to a numpy array
+    faces_array = np.array(valid_faces)
+    resp_objects = [{} for _ in range(len(valid_faces))]
```

Comment:
please do not create resp_objects in advance, instead create resp_object in the loop and append it to resp_objects similar to previous design